inc/fulltext.php

```diff
@@ -279,10 +288,42 @@ function _ft_pageLookup(&$data){
         }
     }
 
+    $pages = _ft_filterResultsByTime($pages);
+
     uksort($pages,'ft_pagesorter');
     return $pages;
 }
 
+
+/**
+ * @param array $results search results in the form pageid => value
+ *
+ * @return array
+ */
+function _ft_filterResultsByTime(array $results) {
+    global $INPUT;
+    if ($INPUT->has('after') || $INPUT->has('before')) {
+        $after = $INPUT->str('after');
+        $after = is_int($after) ? $after : strtotime($after);
+
+        $before = $INPUT->str('before');
+        $before = is_int($before) ? $before : strtotime($before);
+
+        foreach ($results as $id => $value) {
+            $mTime = filemtime(wikiFN($id));
```

Comment:
It's also unfortunate that we run filemtime multiple times now for sorting and filtering. But I guess PHP's statcache should prevent too big performance hits.  wikiFN is cached as well, but I think we could skip the ID cleaning (which isn't cached). I think we can assume that all IDs in the index are clean already.